cli/cli_app.go

```diff
@@ -589,6 +590,9 @@ func needsInit(terragruntOptions *options.TerragruntOptions, terragruntConfig *c
 // Returns true if we need to run `terraform init` to download providers
 func providersNeedInit(terragruntOptions *options.TerragruntOptions) bool {
 	providersPath := util.JoinPath(terragruntOptions.WorkingDir, ".terraform/plugins")
+	if tfDataDir, ok := os.LookupEnv("TF_DATA_DIR"); ok {
+		providersPath = util.JoinPath(tfDataDir, "plugins")
```

Comment:
Should this be `plugins` or `.terraform/plugins`?

Reply:
`plugins`.